.github/workflows/openssl.yml

```diff
@@ -0,0 +1,83 @@
+---
+name: OpenSSL master
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    name: CI with openssl master 
+    runs-on: ubuntu-22.04
+    container: fedora:latest
+    steps:
+      - name: Get Date for DNF cache entry
+        id: get-date-dnf
+        run: |
+          echo "date=$(/bin/date -u "+%Y%V")" >> $GITHUB_OUTPUT
+        shell: bash
+
+      - name: DNF cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            /var/cache/libdnf5
+          key: ${{ runner.os }}-dnf-${{ steps.get-date-dnf.outputs.date }}
+
+      - name: Install Dependencies
+        run: |
+            dnf -y install perl-FindBin perl-IPC-Cmd perl-File-Compare \
+              perl-File-Copy perl-Pod-Html clang git meson cargo expect \
+              pkgconf-pkg-config opensc p11-kit-devel gcc g++ sqlite-devel \
+              python3-six which cmake nss-softokn nss-tools nss-softokn-devel \
+              nss-devel softhsm
```

Comment:
I think this cache should have different key than the one in kryoptic as it caches different DNF packages.

Reply:
Right, they are not the same, fixed in [bc2f565](https://github.com/latchset/pkcs11-provider/pull/550/commits/bc2f5657ede5d81553b14479192e36a93e21cabd).